app/models/user.rb

```diff
@@ -470,6 +477,15 @@ def find_ids(options)
 
   protected
 
+  def promote_to_first_admin
+    account.update({first_admin_id: id})
+  end
+
+  def update_account_first_admin
+    return unless (other_admin = account.admins.where.not(id: id).first)
+    account.update_column(:first_admin_id, other_admin.id) or raise ActiveRecord::ActiveRecordError
```

Comment:
https://github.com/3scale/porta/pull/8#discussion_r237904138
@guicassolato What do you suggest? 😄